internal/wait/value.go

```diff
@@ -75,6 +75,13 @@ func (wv *Value[T]) TryGet() (T, bool) {
 	select {
 	case <-set:
 		wv.mx.Lock()
+
+		if wv.set == nil {
```

Comment:
when waking up after `<-set`, it doesn't mean that nothing happened before the mutex `wv.mx` is locked, there might have been a concurrent call to `.Unset()` which will set the value to nil